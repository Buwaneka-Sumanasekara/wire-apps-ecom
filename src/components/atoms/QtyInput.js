import { useCallback, useEffect, useState } from "react";
import { TextInput, TouchableOpacity, View } from "react-native"
import { Ionicons } from '@expo/vector-icons';

export default QtyInput = ({ onChangeQty }) => {

    const [qty, setQty] = useState(1);

    useEffect(() => {
        onChangeQty(qty);
    }, [qty, onChangeQty])


    const onAddQty = useCallback(() => {
        setQty(qty + 1);
    }, [qty])

    const onDeductQty = useCallback(() => {
        if (qty > 1) {
            setQty(qty - 1);
        }
    }, [qty])

    return (
        <View className={"flex flex-row mr-3"}>
            <TouchableOpacity className={"bg-black p-2 justify-center items-center"} onPress={onDeductQty}>
                <Ionicons name={"add"} size={15} color={"white"} />
            </TouchableOpacity>
            <TextInput editable={false} value={`${qty}`} className={"border border-gray-300  p-2 w-10 text-center  items-center"} />
            <TouchableOpacity className={"bg-black p-2  justify-center  items-center"} onPress={onAddQty} >
                <Ionicons name={"remove"} size={15} color={"white"} />
            </TouchableOpacity>
        </View>
    )
}
