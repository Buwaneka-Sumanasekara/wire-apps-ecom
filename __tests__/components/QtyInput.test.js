import React from 'react';
import renderer from 'react-test-renderer';
import QtyInput from '../../src/components/atoms/QtyInput';

describe('<QtyInput/>', () => {
    it('rendered sucessfully', () => {
        const tree = renderer.create(<QtyInput />).toJSON()
        expect(tree).toMatchSnapshot()
    });
});

