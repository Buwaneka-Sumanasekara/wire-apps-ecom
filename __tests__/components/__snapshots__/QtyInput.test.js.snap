// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QtyInput/> rendered sucessfully 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#000",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <Text />
  </View>
  <TextInput
    editable={false}
    style={
      [
        {
          "borderBottomWidth": 1,
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
        },
        {
          "borderBottomColor": "#d1d5db",
          "borderLeftColor": "#d1d5db",
          "borderRightColor": "#d1d5db",
          "borderTopColor": "#d1d5db",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "width": 40,
        },
        {
          "textAlign": "center",
        },
        {
          "alignItems": "center",
        },
      ]
    }
    value="undefined"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#000",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <Text />
  </View>
</View>
`;
